build/generate_image_tags.sh

```diff
@@ -23,6 +23,10 @@ if [ $(git rev-list -n1 HEAD) == $(git rev-list -n1 origin/master 2>/dev/null) ]
   IMAGE_TAGS+="canary "
 fi
 
+# A "X.Y.Z-canary" image gets built if the current commit is the head of a "origin/release-X.Y.Z" branch.
+# The actual suffix does not matter, only the "release-" prefix is checked.
+IMAGE_TAGS+=$(git branch -r --points-at=HEAD | grep 'origin/release-' | grep -v -e ' -> ' | sed -e 's;.*/release-\(.*\);\1-canary;')
```

Comment:
tested this by checking about a local branch git checkout -b test-release origin/release-0.3. and then running ./build/generate_image_tags.sh. The script output "0.3-canary" tag

Reply:
+1

Are the canary images used in any of the testing/release pipelines? ie we just pull the head images in the staging-head overlay.